### PR TITLE
Use doctr for dev docs

### PR DESCRIPTION
**Pull request by gforsyth**
Uses [doctr](https://github.com/drdoctr/doctr) to build and deploy docs after each PR is merged to `xonsh.github.io/dev`.

Adds in an encrypted deploy key to the `.travis.yml` and also adds `doctr` as a requirement for the `BUILD_DOCS` env.  

For the moment, this is just for dev docs -- note that we won't actually know if it works completely until the PR is merged, since documents aren't deployed otherwise.  
If everything goes as planned, I can add in some logic to also deploy release docs when there's a release tag added, but that can come later.

Also to do: add a link to the dev docs on the main `xon.sh` page (or have a menu for different version of the docs?)

### CODE REVIEW

File: news/doctr.rst [hunk 1]
```diff
@@ -1,6 +1,6 @@
 **Added:**
 
-* Use `doctr` to deploy dev docs to github pages
+* Use ``doctr`` to deploy dev docs to github pages
```
Comment (astronouth7303) [inline news/doctr.rst]: Link?

Comment (gforsyth) [inline news/doctr.rst]: added.  I am bad at promotion

**Comment by scopatz** [comment]
Sweet! I have been waiting for this @gforsyth!

**Comment by scopatz** [comment]
Thanks @gforsyth!

**Comment by gforsyth** [comment]
Right -- I disabled push builds a while ago since Travis was lagging so hard, but `doctr` only pushes after PRs are merged.  

I've reenabled the push builds -- should trigger the next time there's a commit made to master.  And since we now offload some tests to Circle it shouldn't be as much of a problem.

**Comment by gforsyth** [comment]
Ok... so it works and it doesn't.  
When we wrote `doctr` we were targeting docs being served from `gh-pages` branches of the same repo.  So we only deploy to the gh-pages branch.  That said, the organizational github pages sites only support serving from master.  
I'll add support for deploying to any branch, then we'll have to push a new `doctr` release and then this should work as expected.

**Comment by gforsyth** [comment]
Ugh, there's a whole load of problems with having `doctr` deploy to master.  Any objections to just creating a `xonsh_docs` repo that isn't a `github.io` organizational site?

**Comment by scopatz** [comment]
I am fine with it as long as it is `xonsh-docs` :)

**Pull request merged**